java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowReader.java

```diff
@@ -46,7 +46,7 @@
   private VectorLoader loader;
   private VectorSchemaRoot root;
   private Map<Long, Dictionary> dictionaries;
-  private boolean initialized = false;
+  protected boolean initialized = false;
```

Comment:
does this need to be shared?  its strange that both extension of ArrowReader don't use it

Reply:
Yes, as Wes pointed out "If the stream is not empty, and the dictionaries are missing, we should raise the error".  In this case we need loadNextBatch (https://github.com/apache/arrow/pull/4960/files#diff-4d9b9d173e23ff1bedae9331dfcec910R185) to see if it is really empty stream, before this, we need to change this flag, other wise loadNextBatch will throw exception because it need to read schema again which has already been read before.